Core/Core/Credentials/StreamWrapper.cs

```diff
@@ -234,8 +235,20 @@ public async Task<Account> GetAccount()
 
       throw err;
     }
-
-
+    
+    public bool Equals(StreamWrapper wrapper)
```

Comment:
Maybe unnecessary, but could sanitize some of these fields (eg to lowercase) to be 100% sure.